internal/header/feature_tag.go

```diff
@@ -0,0 +1,62 @@
+/* Copyright 2019 Freerware
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package header
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+// FeatureTag represents a feature tag.
+// https://tools.ietf.org/html/rfc2295#section-6.1
+type FeatureTag string
+
+// Equals performss a case-insensitive comparizon with the provided feature tag.
+func (t FeatureTag) Equals(tag FeatureTag) bool {
+	unquoted := strings.ReplaceAll(t.String(), "\"", "")
+	quoted := fmt.Sprintf("%q", t)
+	return strings.ToLower(unquoted) == strings.ToLower(tag.String()) ||
```

Comment:
`EqualFold(...)` is not appropriate for this, as it is exists for Unicode comparison purposes. since feature tags are `US-ASCII`, i made use of `strconv.Unquote(...)` and `strconv.QuoteToASCII(...)` instead!